app/services/sso/apache.rb

```diff
@@ -30,6 +30,7 @@ def support_fallback?
     def authenticated?
       return false unless (self.user = request.env[CAS_USERNAME])
```

Comment:
This extensively uses `request.env` but I wonder why we don't use the [ActionDispatch::Http::Headers](https://api.rubyonrails.org/classes/ActionDispatch/Http/Headers.html) middleware.

Using `request.env` made sense back when we used Passenger, because it could send some things directly to `request.env` meaning users couldn't override them via headers at all. These days we only support Puma behind a reverse proxy and we configure Apache to filter the headers. `CAS_USERNAME` is also hardcoded so it doesn't support using non-headers at all now.

That said, the source code there does have special handling for some headers which includes `Remote-User`. That is good to verify.

Reply:
Do you suggest I should make these changes in this PR or is that somehting for separate PR?

Reply:
This seems like it goes beyond of what I am trying to do, and I would prefer to keep changes here as minimal as possible. I created a [redmine issue](https://projects.theforeman.org/issues/37790) based on your comment, but I would prefer if this change was done in a separate PR.